src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -97,40 +160,60 @@ def _read_hdf5_file(
         else:
             ffspec_file = fsspec_file_system.open(file_path, "rb")
 
-        hdf5_file = h5py.File(name=ffspec_file, mode="r")
+        open_file = h5py.File(name=ffspec_file, mode="r")
 
     elif stream_mode == "ros3":
+        import h5py
+
         assert file_path is not None, "file_path must be specified when using stream_mode='ros3'"
 
         drivers = h5py.registered_drivers()
         assertion_msg = "ROS3 support not enbabled, use: install -c conda-forge h5py>=3.2 to enable streaming"
         assert "ros3" in drivers, assertion_msg
-        hdf5_file = h5py.File(name=file_path, mode="r", driver="ros3")
+        open_file = h5py.File(name=file_path, mode="r", driver="ros3")
 
     elif stream_mode == "remfile":
         import remfile
+        import h5py
 
         assert file_path is not None, "file_path must be specified when using stream_mode='remfile'"
         rfile = remfile.File(file_path)
-        hdf5_file = h5py.File(rfile, "r")
+        open_file = h5py.File(rfile, "r")
 
-    elif file_path is not None:
+    elif stream_mode == "zarr":
+        import zarr
+
+        open_file = zarr.open(file_path, mode="r", storage_options=storage_options)
+
+    elif file_path is not None:  # local
         file_path = str(Path(file_path).resolve())
-        hdf5_file = h5py.File(name=file_path, mode="r")
+        backend = get_backend_from_local_file(file_path)
```

Comment:
This is strange, you have a backend parameter that you don't use but determine from this other function.